statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+
+class SurveyModel(object):
+
+    def __init__(self, design, model_class, init_args={}, fit_args={}):
+        self.design = design
+        self.model = model_class
+        self.init_args = dict(init_args)
+        self.fit_args = dict(fit_args)
+
+    def fit(self, y, X, replicates=None):
+        self.init_args["weights"] = self.design.weights
+        self.params = self._get_params(y, X)
+        if replicates is None:
+            try:
+                k = self.design.nclust
+            except AttributeError:
+                k = self.design.rep_weights.shape[1]
+        else:
+            k = replicates
+        self.replicate_params = []
+        for c in range(k):
+            w = self.design.get_rep_weights(c=c)
+            self.init_args["weights"] = w
+            print('weights', self.init_args['weights'])
+            model = self.model(y, X, **self.init_args)
+            self.replicate_params.append(self._get_params(y, X))
+        self.replicate_params = np.asarray(self.replicate_params)
+        print('new params', self.replicate_params)
+
+        self.replicate_params -= self.params
+
+        # for now, just working with jackknife to see if it works
+        if self.design.cov_method == 'jack':
```

Comment:
A method to get the jackknife multiplier factor? Not sure if I want to do that in SurveyDesign or even SurveyStat if it's only used for the jackknife. I'm thinking to implement a `_get_cov()` method in SurveyModel that is one big function to get `vcov` and is split by `self.design.cov_method` or a bunch of smaller methods depending on cov_method e.g `_get_jack_cov(), _get_linearized_cov()`